modules/calib3d/src/solvepnp.cpp

```diff
@@ -46,12 +46,49 @@
 #include "epnp.h"
 #include "p3p.h"
 #include "ap3p.h"
+#include "ippe.hpp"
 #include "opencv2/calib3d/calib3d_c.h"
 
+//TODO: debug
 #include <iostream>
 
 namespace cv
 {
+static bool isPlanarObjectPoints(InputArray _objectPoints, double threshold, double &val)
+{
+    CV_CheckType(_objectPoints.type(), _objectPoints.type() == CV_32FC3 || _objectPoints.type() == CV_64FC3,
+                 "Type of _objectPoints must be CV_32FC3 or CV_64FC3");
+    Mat objectPoints;
+    if (_objectPoints.type() == CV_32FC3)
+    {
+        _objectPoints.getMat().convertTo(objectPoints, CV_64F);
+    }
+    else
+    {
+        objectPoints = _objectPoints.getMat();
+    }
+
+    Scalar meanValues = mean(objectPoints);
+    int nbPts = objectPoints.checkVector(3, CV_64F);
+    Mat objectPointsCentred(nbPts, 3, CV_64FC1);
+    for (int i = 0; i < objectPointsCentred.rows; i++)
+    {
+        objectPointsCentred.at<double>(i,0) = objectPoints.ptr<Vec3d>(0)[i](0) - meanValues(0);
+        objectPointsCentred.at<double>(i,1) = objectPoints.ptr<Vec3d>(0)[i](1) - meanValues(1);
+        objectPointsCentred.at<double>(i,2) = objectPoints.ptr<Vec3d>(0)[i](2) - meanValues(2);
+    }
+
+    Mat w, u, vt;
+    SVDecomp(objectPointsCentred, w, u, vt);
+    val = w.at<double>(2)/w.at<double>(1);
+    return (w.at<double>(2)/w.at<double>(1) < threshold);
```

Comment:
reuse `val`?
- fabs()?
- `a / b < c`   ==>  `a < c * b`?

Reply:
I have used the multiplication.
I don't know exactly what this quantity represents exactly.
fabs is not used here :
https://github.com/opencv/opencv/blob/d0032b07178ffde4f13ed4878744b5a8845f16b1/modules/calib3d/src/calibration.cpp#L1112

and here
https://github.com/tobycollins/IPPE/blob/3af0127a30fcb0d35fa2a34f4abd93278e631621/cpp/src/IPPE/ippe.cpp#L1037